systemverilog-plugin/Makefile

```
@@ -27,31 +27,23 @@ SOURCES = UhdmAst.cc \
 # Directory to search for Surelog and UHDM libraries
 UHDM_INSTALL_DIR ?= /usr/local
 
+# Tell pkg-config to look in the provided install path first.
+# PKG_CONFIG_PATH and PKG_CONFIG_PATH_FOR_TARGET are search paths it looks in
+# so set the environment variables and prefix with our local install first
+PKG_CONFIG_INVOKE=\
+   PKG_CONFIG_PATH=$(UHDM_INSTALL_DIR)/lib/pkconfig:${PKG_CONFIG_PATH} \
+   PKG_CONFIG_PATH_FOR_TARGET=$(UHDM_INSTALL_DIR)/lib/pkconfig:${PKG_CONFIG_PATH_FOR_TARGET} \
+   pkg-config
+
 include ../Makefile_plugin.common
 
+# A litmus-test: make compilation fail if pkg-config fails
+$(NAME).so:
+	$(PKG_CONFIG_INVOKE) --cflags Surelog
+
```

This is only called when make builds the default target (`all`). [Targets like `install` don't invoke it][install], even when they have to build the lib. [`$(NAME).so` is a phony target][nameso] which exists only for backward compatibility. [The actual file, `${_SO_LIB}`][solib], is now inside `$(BUILD_DIR)`.

One solution I can think of:
```suggestion
.SECONDARY: $(BUILD_DIR)/.$(NAME)-deps-test
$(BUILD_DIR)/.$(NAME)-deps-test:
	$(PKG_CONFIG_INVOKE) --cflags Surelog

${_SO_LIB}: | $(BUILD_DIR)/.$(NAME)-deps-test

```

Since `_SO_LIB` will no longer be an internal variable, please also rename it to `SO_LIB`.

[nameso]: https://github.com/chipsalliance/yosys-f4pga-plugins/blob/c1ece1c6d7d2287d90ff515043ea82e7e1ad7323/Makefile_plugin.common#L110-L111
[solib]: https://github.com/chipsalliance/yosys-f4pga-plugins/blob/c1ece1c6d7d2287d90ff515043ea82e7e1ad7323/Makefile_plugin.common#L104
[install]: https://github.com/antmicro/yosys-systemverilog/actions/runs/4563824062/jobs/8052801038#step:10:4385

---

Done. Renamed _SO_LIB -> SO_LIB in a separate commit.